src/Libraries/Microsoft.Extensions.Options.Contextual/ContextualOptionsServiceCollectionExtensions.cs

```diff
@@ -54,108 +56,71 @@ public static IServiceCollection AddContextualOptions(this IServiceCollection se
     /// <returns>The value of <paramref name="services"/>.</returns>
     public static IServiceCollection Configure<TOptions>(
         this IServiceCollection services,
-        string name,
+        string? name,
         Func<IOptionsContext, CancellationToken, ValueTask<IConfigureContextualOptions<TOptions>>> loadOptions)
         where TOptions : class
         => services
             .AddContextualOptions()
             .AddSingleton<ILoadContextualOptions<TOptions>>(
                 new LoadContextualOptions<TOptions>(
-                    Throw.IfNull(name),
+                    name,
                     Throw.IfNull(loadOptions)));
 
     /// <summary>
     /// Registers an action used to configure a particular type of options.
     /// </summary>
     /// <typeparam name="TOptions">The options type to be configured.</typeparam>
     /// <param name="services">The <see cref="IServiceCollection"/> to add the services to.</param>
-    /// <param name="configureOptions">The action used to configure the options.</param>
+    /// <param name="configure">The action used to configure the options.</param>
     /// <returns>The value of <paramref name="services"/>.</returns>
-    public static IServiceCollection Configure<TOptions>(this IServiceCollection services, Action<IOptionsContext, TOptions> configureOptions)
+#pragma warning disable S3872 // Parameter names should not duplicate the names of their methods
+    public static IServiceCollection Configure<TOptions>(this IServiceCollection services, Action<IOptionsContext, TOptions> configure)
+#pragma warning restore S3872 // Parameter names should not duplicate the names of their methods
         where TOptions : class
-        => services.Configure(Options.Options.DefaultName, Throw.IfNull(configureOptions));
+        => services.Configure(Options.Options.DefaultName, Throw.IfNull(configure));
 
     /// <summary>
     /// Registers an action used to configure a particular type of options.
     /// </summary>
     /// <typeparam name="TOptions">The options type to be configured.</typeparam>
     /// <param name="services">The <see cref="IServiceCollection"/> to add the services to.</param>
     /// <param name="name">The name of the options to configure.</param>
-    /// <param name="configureOptions">The action used to configure the options.</param>
+    /// <param name="configure">The action used to configure the options.</param>
     /// <returns>The value of <paramref name="services"/>.</returns>
-    public static IServiceCollection Configure<TOptions>(this IServiceCollection services, string name, Action<IOptionsContext, TOptions> configureOptions)
+#pragma warning disable S3872 // Parameter names should not duplicate the names of their methods
```

Comment:
Is there a reason not to call it `configureOptions`?

Reply:
feedback provided by @geeknoid, to align the parameter names with Options -> but options also uses configureOptions -> https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.Extensions.Options/src/OptionsServiceCollectionExtensions.cs

@geeknoid could you confirm if I can revert this change?